File: src/main/graphql/io/github/graphql/model/UserContentEditTO.kt
package io.github.graphql.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import java.util.StringJoiner

@javax.annotation.Generated(
    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
    date = "2021-03-05T19:09:00+0800"
)
open class UserContentEditTO(
    val createdAt: String,
    val deletedAt: String?,
    val deletedBy: ActorTO?,
    val diff: String?,
    val editedAt: String,
    val editor: ActorTO?,
    override
    val id: String,
    val updatedAt: String
) : NodeTO {

    // In the future, it maybe change.
    override fun toString(): String {
        val joiner = StringJoiner(", ", "{ ", " }")
        joiner.add("createdAt: " + GraphQLRequestSerializer.getEntry(createdAt))
        if (deletedAt != null) {
            joiner.add("deletedAt: " + GraphQLRequestSerializer.getEntry(deletedAt))
        }
        if (deletedBy != null) {
            joiner.add("deletedBy: " + GraphQLRequestSerializer.getEntry(deletedBy))
        }
        if (diff != null) {
            joiner.add("diff: " + GraphQLRequestSerializer.getEntry(diff))
        }
        joiner.add("editedAt: " + GraphQLRequestSerializer.getEntry(editedAt))
        if (editor != null) {
            joiner.add("editor: " + GraphQLRequestSerializer.getEntry(editor))
        }
        joiner.add("id: " + GraphQLRequestSerializer.getEntry(id))
        joiner.add("updatedAt: " + GraphQLRequestSerializer.getEntry(updatedAt))
        return joiner.toString()
    }
}

File: src/main/scala-graphql/io/github/graphql/model/UserContentEditTO.scala
package io.github.graphql.model

import com.kobylynskyi.graphql.codegen.model.graphql.GraphQLRequestSerializer
import scala.collection.JavaConverters._

@javax.annotation.Generated(
    value = Array("com.kobylynskyi.graphql.codegen.GraphQLCodegen"),
    date = "2021-03-06T19:48:00+0800"
)
class UserContentEditTO(
    @javax.validation.constraints.NotNull
    createdAt: String,
    deletedAt: String,
    deletedBy: ActorTO,
    diff: String,
    @javax.validation.constraints.NotNull
    editedAt: String,
    editor: ActorTO,
    @javax.validation.constraints.NotNull
    override val id: String,
    @javax.validation.constraints.NotNull
    updatedAt: String
) extends NodeTO {

    override def toString(): String = {
        Seq(
            if (createdAt != null) "createdAt: " + GraphQLRequestSerializer.getEntry(createdAt) else "",
            if (deletedAt != null) "deletedAt: " + GraphQLRequestSerializer.getEntry(deletedAt) else "",
            if (deletedBy != null) "deletedBy: " + GraphQLRequestSerializer.getEntry(deletedBy) else "",
            if (diff != null) "diff: " + GraphQLRequestSerializer.getEntry(diff) else "",
            if (editedAt != null) "editedAt: " + GraphQLRequestSerializer.getEntry(editedAt) else "",
            if (editor != null) "editor: " + GraphQLRequestSerializer.getEntry(editor) else "",
            if (id != null) "id: " + GraphQLRequestSerializer.getEntry(id) else "",
            if (updatedAt != null) "updatedAt: " + GraphQLRequestSerializer.getEntry(updatedAt) else ""
        ).filter(_ != "").mkString("{", ",", "}")
    }
}

object UserContentEditTO {

    def builder(): UserContentEditTO.Builder = new Builder()

    class Builder {

        private var createdAt: String = _
        private var deletedAt: String = _
        private var deletedBy: ActorTO = _
        private var diff: String = _
        private var editedAt: String = _
        private var editor: ActorTO = _
        private var id: String = _
        private var updatedAt: String = _

        def setCreatedAt(createdAt: String): Builder = {
            this.createdAt = createdAt
            this
        }

        def setDeletedAt(deletedAt: String): Builder = {
            this.deletedAt = deletedAt
            this
        }

        def setDeletedBy(deletedBy: ActorTO): Builder = {
            this.deletedBy = deletedBy
            this
        }

        def setDiff(diff: String): Builder = {
            this.diff = diff
            this
        }

        def setEditedAt(editedAt: String): Builder = {
            this.editedAt = editedAt
            this
        }

        def setEditor(editor: ActorTO): Builder = {
            this.editor = editor
            this
        }

        def setId(id: String): Builder = {
            this.id = id
            this
        }

        def setUpdatedAt(updatedAt: String): Builder = {
            this.updatedAt = updatedAt
            this
        }

        def build(): UserContentEditTO = new UserContentEditTO(createdAt, deletedAt, deletedBy, diff, editedAt, editor, id, updatedAt)

    }
}
